src/Compilers/CSharp/Portable/Symbols/Source/SourceConstructorSymbolBase.cs

```diff
@@ -24,7 +24,11 @@ internal abstract class SourceConstructorSymbolBase : SourceMemberMethodSymbol
             DiagnosticBag diagnostics)
             : base(containingType, syntax.GetReference(), ImmutableArray.Create(location), SyntaxFacts.HasYieldOperations(syntax))
         {
-            Debug.Assert(syntax.IsKind(SyntaxKind.ConstructorDeclaration) || syntax.IsKind(SyntaxKind.ClassDeclaration) || syntax.IsKind(SyntaxKind.StructDeclaration));
+            Debug.Assert(
+                syntax.IsKind(SyntaxKind.ConstructorDeclaration) ||
+                syntax.IsKind(SyntaxKind.ClassDeclaration) ||
```

Comment:
>syntax.IsKind(SyntaxKind.ClassDeclaration) [](start = 16, length = 42)

It feels like we should remove class and struct, i.e. replace them with record. Similarly in other places

Reply:
This was the only place I found, but I can fix up more places if you see them.